templates/courseware/courseware.html

```diff
@@ -0,0 +1,221 @@
+  <script type="text/javascript" src="${static.url('js/vendor/jquery.scrollTo-1.4.2-min.js')}"></script>
+  <script type="text/javascript" src="${static.url('js/vendor/flot/jquery.flot.js')}"></script>
+
+  ## codemirror
+  <script type="text/javascript" src="${static.url('js/vendor/codemirror-compressed.js')}"></script>
+
+  <%static:js group='courseware'/>
+  <%static:js group='discussion'/>
+
+  <%include file="../discussion/_js_body_dependencies.html" />
+  % if staff_access:
+  	<%include file="xqa_interface.html"/>
+  % endif
+
+  <script type="text/javascript">
+    var $$course_id = "${course.id | escapejs}";
+  </script>
+
+% if timer_expiration_duration:
+  <script type="text/javascript">
+    var timer = {
+      timer_inst : null,
+      end_time : null,
+      get_remaining_secs : function(endTime) {
+        var currentTime = new Date();
+        var remaining_secs = Math.floor((endTime - currentTime)/1000);
+        return remaining_secs;
+      },
+      get_time_string : function() {
+        function pretty_time_string(num) {
+          return ( num < 10 ? "0" : "" ) + num;
+        }
+        // count down in terms of hours, minutes, and seconds:
+        var hours = pretty_time_string(Math.floor(remaining_secs / 3600));
+        remaining_secs = remaining_secs % 3600;
+        var minutes = pretty_time_string(Math.floor(remaining_secs / 60));
+        remaining_secs = remaining_secs % 60;
+        var seconds = pretty_time_string(Math.floor(remaining_secs));
+
+        var remainingTimeString = hours + ":" + minutes + ":" + seconds;
+        return remainingTimeString;
+      },
+      update_time : function(self) {
+        remaining_secs = self.get_remaining_secs(self.end_time);
+        if (remaining_secs <= 0) {
+          self.end(self);
+        }
+        $('#exam_timer').text(self.get_time_string(remaining_secs));
+      },
+      start : function() { var that = this;
+        // set the end time when the template is rendered.
+        // This value should be UTC time as number of milliseconds since epoch.
+        this.end_time = new Date((new Date()).getTime() + ${timer_expiration_duration});
+        this.timer_inst = setInterval(function(){ that.update_time(that); }, 1000);
+      },
+      end : function(self) {
+        clearInterval(self.timer_inst);
+        // redirect to specified URL:
+        window.location = "${time_expired_redirect_url}";
+      }
+    }
+    // start timer right away:
+    timer.start();
+  </script>
+% endif
+
+% if show_chat:
+  <script type="text/javascript" src="${static.url('js/vendor/candy_libs/libs.min.js')}"></script>
+  <script type="text/javascript" src="${static.url('js/vendor/candy.min.js')}"></script>
+
+  <script type="text/javascript">
+    // initialize the Candy.js plugin
+    $(document).ready(function() {
+      Candy.init("http://${chat['domain']}:5280/http-bind/", {
+        core: { debug: true, autojoin: ["${chat['room']}@conference.${chat['domain']}"] },
```

Comment:
So this one is from the default template for EdX - I didn't write that bit.  I'm not sure we'll ever use it, given that it looks like it's for course chat